MissingNum.java

```diff
@@ -0,0 +1,44 @@
+// Time Complexity : O(n) where n is size of array
+// Space Complexity :O(n)
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : No
+
+
+// Your code here along with comments explaining your approach
+
+class Solution {
+    public List<Integer> findDisappearedNumbers(int[] nums) {
+        List<Integer> missingNumbers = new ArrayList<Integer>(); //to store missing numbers
+        HashSet<Integer> numbers = new HashSet<Integer>(); // To store numbers
+        for(int i : nums){
+            numbers.add(i); // takes care of duplicate values.
+        }
+        
+        for(int i = 1; i <=nums.length; i++){
+            if(!numbers.contains(i)){
+                missingNumbers.add(i);
+            }
+        }
+        return missingNumbers;
+    }
+}
+
+// with no extra space O(1)
+
```

Comment:
Here the approach is, we iterate through the input array and mark elements as negative using nums[nums[i]-1] = -nums[nums[i]-1]. In this way, all the numbers that we have visited will be marked as negative. In the second iteration, if a value is not marked as negative, it implies we have never visited that index before, so just add it to the return list.